File: CS/VM-DrivenWizard/ViewModels/MainWindowViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Threading.Tasks;


namespace VM_DrivenWizard.ViewModels {
    public class MainWindowViewModel : ViewModelBase {
        public virtual string Text { get; protected set; }
        IDialogService DialogService => this.GetService<IDialogService>();

        [Command]
        public void ShowDialog() {
            var dialogWindowViewModel = new DialogWindowViewModel();
            var wizardResult = DialogService.ShowDialog(MessageButton.OKCancel, "Wizard", dialogWindowViewModel).ToString();
            string selectedSong = dialogWindowViewModel.GetSelectedSong();
            Text = $"Wizard result: {wizardResult}{(string.IsNullOrEmpty(selectedSong) ? string.Empty : $", you choose {selectedSong}")}";
        }
    }
}

File: VB/VM-DrivenWizard/ViewModels/MainWindowViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations

Namespace VM_DrivenWizard.ViewModels

    Public Class MainWindowViewModel
        Inherits ViewModelBase

        Public Overridable Property Text As String

        Private ReadOnly Property DialogService As IDialogService
            Get
                Return GetService(Of IDialogService)()
            End Get
        End Property

        <Command>
        Public Sub ShowDialog()
            Dim dialogWindowViewModel = New DialogWindowViewModel()
            Dim wizardResult = DialogService.ShowDialog(MessageButton.OKCancel, "Wizard", dialogWindowViewModel).ToString()
            Dim selectedSong As String = dialogWindowViewModel.GetSelectedSong()
            Text = $"Wizard result: {wizardResult} {If(String.IsNullOrEmpty(selectedSong), String.Empty, $", you choose {selectedSong}")}"
        End Sub
    End Class
End Namespace
